server/data/tokenStore.test.ts

```diff
@@ -1,34 +1,66 @@
-import { RedisClient } from 'redis'
+import nock from 'nock'
+import redis, { RedisClient } from 'redis-mock'
 import TokenStore from './tokenStore'
+import config from '../config'
 
-const redisClient = {
-  on: jest.fn(),
-  get: jest.fn(),
-  set: jest.fn(),
-}
+let tokenStore: TokenStore
+let redisClient: RedisClient
 
-describe('tokenStore', () => {
-  let tokenStore: TokenStore
+jest.mock('redis', () => redis)
 
-  beforeEach(() => {
-    tokenStore = new TokenStore(redisClient as unknown as RedisClient)
-  })
+jest.mock('../authentication/clientCredentials', () => {
+  return jest.fn().mockImplementation(() => 'Basic clientId:secret')
+})
 
-  afterEach(() => {
-    jest.resetAllMocks()
+beforeEach(() => {
+  redisClient = redis.createClient()
+  tokenStore = new TokenStore()
+})
+
+afterEach(done => {
+  redisClient.flushall(done)
+  jest.clearAllMocks()
+  nock.cleanAll()
+})
+
+describe('Token Store', () => {
+  it('Token for user is returned from redis is exists', async () => {
+    redisClient.set('joebloggs', 'saved token')
+
+    const token = await tokenStore.getAuthToken('joebloggs')
+
+    expect(token).toEqual('saved token')
   })
 
-  it('Can retrieve token', async () => {
-    redisClient.get.mockImplementation((key, callback) => callback(null, 'token-1'))
+  it('Auth is called to get and save a token for user if one is not saved', async () => {
+    nock(config.apis.hmppsAuth.url, {
```

Comment:
I see the issue - normal users do not have client-credential accounts - and we don't call out to get them another token. Only system-api clients have these accounts. This is why some client methods (not many - about 3 or 4) accept the user's own token, held res.locals.user, and passed through to the API client. All the rest will get a system token, from redis, or if absent there, from auth and store it.